### PR TITLE
Pulls and renders categories in left menu

### PR DESCRIPTION
**Pull request by calisio**
Adds new feature that pulls categories from the database and renders these categories and its links in the left menu.

### CODE REVIEW

**Comment by nokierae** [comment]
Thanks @calisio ! I'll leave some feedback later this afternoon.

**Review by PThorpe92 — APPROVED** [review]

**Comment by nokierae** [comment]
Update: I ran out of time tonight, so I'll have to leave feedback tomorrow.

File: resources/js/Components/LeftMenu.tsx [hunk 1]
```diff
@@ -1,11 +1,70 @@
+import axios from "axios";
 import Brand from "./Brand";
 import {
     HomeIcon,
     ArchiveBoxIcon,
     BookOpenIcon,
 } from "@heroicons/react/24/solid";
+import useSWR from "swr";
+
+interface CategoryParameters {
```
Comment (nokierae) [inline resources/js/Components/LeftMenu.tsx:10]: I would move these types to the resources/js/common.ts file and rename them to just Category and CategoryLink.

File: resources/js/Components/LeftMenu.tsx [hunk 2]
```diff
@@ -1,11 +1,70 @@
...
+                    {categoryName}
+                </summary>
+                <ul>{linksList}</ul>
+            </details>
+        </li>
+    );
+}
+
+function getCategoryItems() {
+    const { data, error, isLoading } = useSWR("/api/v1/categories", (url) =>
```
Comment (nokierae) [inline resources/js/Components/LeftMenu.tsx:42]: Technically, all hooks need to be in the body of the component and shouldn't be placed in functions. I would moev useSWR into the LeftMenu component and pass the data into this function.

**Pull request merged**